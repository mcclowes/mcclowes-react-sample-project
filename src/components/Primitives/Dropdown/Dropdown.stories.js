import Dropdown from "./DropdownContainer";
import React from "react";

export default { title: "Dropdown" };

export const empty = () => {
  return <Dropdown />;
};

export const withOptions = () => {
  return <Dropdown placeholder="Select..." options={[1, 2, 3, 4, 5]} />;
};

export const withCallback = () => {
  return (
    <Dropdown
      placeholder="Select..."
      options={[1, 2, 3, 4, 5]}
      callback={(value) => alert("You clicked " + value)}
    />
  );
};
