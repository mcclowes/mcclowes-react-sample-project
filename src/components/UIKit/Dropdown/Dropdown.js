import Icon from "../Icon";
import PropTypes from "prop-types";
import React from "react";

import {
  DropdownWrapper,
  DropdownPlaceholder,
  DropdownOptions,
  DropdownOption,
} from "./csx";

const Dropdown = (props) => {
  const { placeholder, options, open, setOpen, value, handleClick } = props;

  return (
    <DropdownWrapper onClick={() => setOpen(!open)}>
      <DropdownPlaceholder value={value} open={open}>
        {value === undefined ? placeholder : value}

        <Icon icon="DOWN" />
      </DropdownPlaceholder>

      {options && (
        <DropdownOptions open={open}>
          {options.map((option, i) => {
            return (
              <DropdownOption
                key={i}
                onClick={() => handleClick(option)}
                selected={value === option}
              >
                {option}
              </DropdownOption>
            );
          })}
        </DropdownOptions>
      )}
    </DropdownWrapper>
  );
};

Dropdown.propTypes = {
  handleClick: PropTypes.func,
  open: PropTypes.bool,
  options: PropTypes.array,
  placeholder: PropTypes.string,
  setOpen: PropTypes.func,
  value: PropTypes.any,
};

export default Dropdown;
