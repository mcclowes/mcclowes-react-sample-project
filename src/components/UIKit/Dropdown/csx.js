import styled from "styled-components/macro";

export const DropdownPlaceholder = styled.div`
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ${(props) =>
    props.value === undefined ? `color: ${props.theme.colors.gray};` : ""};
`;

DropdownPlaceholder.defaultProps = {
  theme: {
    colors: {
      gray: "#aaaaaa",
    },
  },
};

export const DropdownOptions = styled.div`
  ${(props) =>
    props.open
      ? `border-top: 1px solid ${props.theme.colors.gray};`
      : "display:none;"};
`;

DropdownOptions.defaultProps = {
  theme: {
    colors: {
      gray: "#aaaaaa",
    },
  },
};

export const DropdownOption = styled.div``;

export const DropdownWrapper = styled.div`
  border: 1px solid ${(props) => props.theme.colors.gray};
  border-radius: ${(props) => props.theme.sizes.border.radius};
  cursor: pointer;
  background: ${(props) => props.theme.colors.white};

  ${DropdownPlaceholder} {
    svg {
      fill: ${(props) => props.theme.colors.gray};
      transform: ${(props) => (props.open ? "rotate(180deg);" : "")};
    }
  }

  ${DropdownOption} {
    padding: 0.25em 0.5em;

    &:hover {
      background-color: ${(props) => props.theme.colors.lightgray};
    }

    &:active {
      background-color: ${(props) => props.theme.colors.gray};
    }
  }
`;

DropdownWrapper.defaultProps = {
  theme: {
    colors: {
      white: "#ffffff",
      gray: "#aaaaaa",
      lightgray: "#eeeeee",
    },
    sizes: {
      border: {
        radius: "3px",
      },
    },
  },
};
