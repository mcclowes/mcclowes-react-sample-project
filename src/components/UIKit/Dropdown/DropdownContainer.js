import Dropdown from "./Dropdown";
import PropTypes from "prop-types";
import React, { useState } from "react";

const DropdownContainer = (props) => {
  const [open, setOpen] = useState(false);
  const [value, setValue] = useState(undefined);

  const { callback } = props;

  const handleClick = (value) => {
    setValue(value);
    if (callback) {
      callback(value);
    }
  };

  return (
    <Dropdown
      value={value}
      handleClick={handleClick}
      open={open}
      setOpen={setOpen}
      {...props}
    />
  );
};

DropdownContainer.propTypes = {
  callback: PropTypes.func,
};

export default DropdownContainer;
