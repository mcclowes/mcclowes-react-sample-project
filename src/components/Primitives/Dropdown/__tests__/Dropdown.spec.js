import Dropdown from "../DropdownContainer";
import React from "react";
import { shallow } from "enzyme";

describe("Dropdown", () => {
  it("renders component", () => {
    const wrapper = shallow(<Dropdown placeholder="Default Dropdown" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders component with options", () => {
    const wrapper = shallow(
      <Dropdown placeholder="Default Dropdown" options={[1, 2, 3, 4, 5]} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
