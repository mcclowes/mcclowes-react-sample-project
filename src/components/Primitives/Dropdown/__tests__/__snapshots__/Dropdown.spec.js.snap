// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders component 1`] = `
<Dropdown
  handleClick={[Function]}
  open={false}
  placeholder="Default Dropdown"
  setOpen={[Function]}
/>
`;

exports[`Dropdown renders component with options 1`] = `
<Dropdown
  handleClick={[Function]}
  open={false}
  options={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  placeholder="Default Dropdown"
  setOpen={[Function]}
/>
`;
